donut/modules/editor/static/script.js

```diff
@@ -0,0 +1,197 @@
+    }
+  });
+}
+
+//###########
+
+function modHighlightedString(string1, string2)
+{
+  var ta = $('#source')[0];
+  if (ta.selectionStart === ta.selectionEnd)
+  {
+    insert(string2, false);
+  }
+  else {
+    insert(string1, true);
+  }
+}
+
+// All markdown related functions
+function insert_heading(event){
+  var size = event.data.size;
+  var string = "";
+  for(var i = 0; i < size; i++)
+  {
+    string += "#";
+  }
+  modHighlightedString(string, string + "Heading_title" + string+ "\n");
+}
+
+function insert_italic(){
+  modHighlightedString("*", "*italicText*");
+}
+
+function insert_bold(){
+  modHighlightedString("**", "**boldText**");
+}
+
+function insert_link(){
+  insert("[Link title](example.com)", false);
+}
+
+function insert_image(){
+  insert("![Alt text](url/to/image)", false);
+}
+
+function insert_ulist(){
+  insert("* list \n", false);
+}
+
+function insert_olist(){
+  insert("1. list \n", false);
+}
+
+
+function insert(string, selected){
+  var ta = $("#source")[0];
+  var begin = ta.selectionStart;
+  var end = ta.selectionEnd;
+  if(selected)
+  {
+    if (document.queryCommandSupported('insertText')) {
+      ta.setSelectionRange(begin, begin);
+      ta.focus();
+      document.execCommand('insertText', false, string);
+      ta.setSelectionRange(end + string.length, end + string.length);
+      ta.focus();
+      document.execCommand('insertText', false, string);
+    } else{
+      ta.value = ta.value.substring(0, begin)
+      + string + ta.value.substring(begin, end) + string
+      + ta.value.substring(end);
+    }
+  }
+  else {
+    if (document.queryCommandSupported('insertText')) {
+      document.execCommand('insertText', false, string);
+    } else{
+      ta.value = ta.value.substring(0, begin)
+      + string + ta.value.substring(begin);
+    }
+  }
+}
+
+// Saves the page created
+function save(){
+  // Construct the html and get the info needed
+  var text = $("#source").val();
+  var title = $('#text_title').val();
+  // Checking for valid titles
+  if (title === '')
+  {
+    window.alert("Enter a title for your new page!");
+    return;
+  }
+  $.ajax({
+    url: '/pages/_check_errors',
+    type: 'POST',
+    data:{markdown:text, title:title},
+    success: function(data) {
+      if (data.error === "Duplicate title" || data.error === "")
+      {
+        if (data.error === "Duplicate title") {
+          var res = confirm("You are overriding an existing file!");
+        }
+        else {
+          var res = true;
+        }
+        if (res){
+          $.ajax({
+            url: '/pages/_save',
+            type: 'POST',
+            data:{markdown:text, title:title},
+            success: function(data) {
+              window.location.href = data.url;
+            },
+            error: function(data){
+              window.alert("Invalid permissions");
+            }
+          });
+        }
+      } else {
+        window.alert(data.error);
+      }
```

Comment:
Don't think it really makes sense to `alert('Duplicate title')` if the user clicks "no" on the `confirm` dialog

Reply:
The many brackets make it hard to read but the data.error is for when it is not `Duplicate title` or ``

Reply:
Ah, makes sense. Sorry I misread it